### DISCUSSION TITLE
Verification of the yaw angles returned by optimize_yaw_angles method

### DISCUSSION THREAD

**Discussion by fwasilcz** [NatLabRockies/floris#1014] [Q&A]
Hi,
I was running a yaw optimization problem for tens of turbines for various layouts and got a bunch of wierd results. 
For most wind directions the optimizer works fine, improving the power output. However, there are selected wind directions (one or more, depending on the layout), where applying the angles from the optimizer cause the decrease in power generated. Looking at the layouts and flow fileds it kind of makes sense - those are the WDs where the wake interactions are difficult to avoid for larger cluster. But then the optimal angles from the optimizer should reflect that, by returing 0 as optimial angle instead of some other values that decrease the power output. I expected this may have been done through "verify_convergence", but it is not working (I'll post about it separately, not to make too much mess since it is a tangential topic). 
Could you comment if the yaw angles from the optimizer are somehow evaluated before they are returned as optimal?
Regards
Filip

**Comment by misi9170** [comment]
Hi Filip,

I've replied to your issue on the `verify_convergence` bug over on #1015 , but for completeness, [here is the procedure run when `verify_convergence=True`](https://github.com/NREL/floris/blob/main/floris/optimization/yaw_optimization/yaw_optimization_base.py#L420).

I'm not sure whether the issue you seem to be having ("optimal" nonzero yaw angles that would in fact be better left as zero) is to do with running the yaw optimization with `ParallelFlorisModel.optimize_yaw_angles()`, or whether it is coming from something further down in the code. Have you tried running your optimization (or a subset of it) in serial using `YawOptimizationSR` directly, by any chance? It would be great if you could see if that still causes the problem.

Best,
Misha

**Reply by fwasilcz** [reply]
Hello Misha, 
Thank you very much for your reply! It seems that the faulty "verify convergence" may indeed be an issue. I'll test new parallel simulation class and case is serial mode and get back to you.
Regards
Filip
